app/env_canada.py

```diff
@@ -1,77 +1,215 @@
+    if os.path.exists(LOGGING_CONFIG):
+        with open(LOGGING_CONFIG) as config_file:
+            CONFIG = json.load(config_file)
+        logging.config.dictConfig(CONFIG)
+
+logger = logging.getLogger(__name__)
+
+
+class UnhadeledPredictionModelType(Exception):
+    """ Exception raised when an unknown model type is encountered. """
+
+
+# pylint: disable=too-many-locals
+def parse_env_canada_filename(filename):
+    """ Take a grib filename, as per file name nomenclature defined at
+    https://weather.gc.ca/grib/grib2_glb_25km_e.html, and parse into a meaningful object.
+    """
+
+    base = os.path.basename(filename)
+    parts = base.split('_')
+    model = parts[1]
+    variable = parts[2]
+    level_type = parts[3]
+    level = parts[4]
+    variable_name = '_'.join(
+        [variable, level_type, level])
+    projection = parts[5]
+    forecast_start = parts[6][:-2]
+    run_time = parts[6][-2:]
+    model_run_timestamp = datetime.datetime(
+        year=int(forecast_start[:4]),
+        month=int(forecast_start[4:6]),
+        day=int(forecast_start[6:8]),
+        hour=int(run_time), tzinfo=pytz.utc)
+    last_part = parts[7].split('.')
+    forecast_hour = last_part[0][1:]
+    forecast_timestamp = model_run_timestamp + \
+        datetime.timedelta(hours=int(forecast_hour))
+
+    if model == 'glb':
+        model_abbreviation = 'GDPS'
+    elif model == 'reg':
+        model_abbreviation = 'RDPS'
+    else:
+        raise UnhadeledPredictionModelType(
+            'Unhandeled prediction model type found', model)
+
+    info = ModelRunInfo()
+    info.model_abbreviation = model_abbreviation
+    info.projection = projection
+    info.model_run_timestamp = model_run_timestamp
+    info.forecast_timestamp = forecast_timestamp
+    info.variable_name = variable_name
+    return info
+
+
+def get_file_date_part(now, hour) -> str:
+    """ Construct the part of the filename that contains the model run date
+    """
+    if now.hour < hour:
+        # if now (e.g. 10h00) is less than model run (e.g. 12), it means we have to look for yesterdays
+        # model run.
+        day = now.day - 1
+    else:
+        day = now.day
+    date = '{year}{month:02d}{day:02d}'.format(
+        year=now.year, month=now.month, day=day)
+    return date
+
+
+def get_utcnow():
+    """ Wrapped datetime.datetime.uctnow() for easier mocking in unit tests.
+    """
+    return datetime.datetime.utcnow()
 
 
 def get_download_urls():
     """ Create a list of urls to download and return it """
-    urls = []
-    date = tz.localize(datetime.now()).strftime('%Y%m%d')  # YYYYMMDD
+    # We always work in UTC:
+    now = get_utcnow()
 
     # hh: model run start, in UTC [00, 12]
     # hhh: forecast hour [000, 003, 006, ..., 240]
-    for hh in ['00', '12']:
+    # pylint: disable=invalid-name
+    for hour in [0, 12]:
+        hh = '{:02d}'.format(hour)
         for h in range(81):
```

Comment:
why 81? I can't tell what this number relates to

Reply:
3 hour intervals, up to 240 hours into the future (3*80 == 240); I've added a comment, and changed the code to rather read as follows:
```
# For the global model, we have prediction at 3 hour intervals up to 240 hours.
        for h in range(0, 241, 3):
```